shed/event_streams.py

```diff
@@ -894,7 +894,7 @@ def event(self, doc):
         else:
             data[self.state_key] = self.state
             try:
-                result = self.func(data)
+                result = self.func(**data)
```

Comment:
Should we take in something more args friendly? (not all things take kwargs `operator.add`)

Reply:
that was laziness on my part. I agree. I would vote for the API to be similar to `streamz`, `func(prevstate, newstate)`. I updated this.
Also, the `streamz` library goes further to return both the previous state and the incoming state:
```python
newstate, newelem = func(curstate, incomingelem)
```

We could consider using this before the API is used more often